tools/run_tests/performance/loadtest_config.py

```diff
@@ -94,34 +107,62 @@ def gen_run_indices(runs_per_test: int) -> Iterable[str]:
         yield prefix_fmt.format(i)
 
 
-def gen_loadtest_configs(base_config: yaml.YAMLObject,
-                         scenarios: Iterable[Mapping[str, Any]],
-                         loadtest_name_prefix: str,
-                         uniquifiers: Iterable[str],
-                         annotations: Mapping[str, str],
-                         runs_per_test: int = 1) -> Iterable[yaml.YAMLObject]:
-    """Generates LoadTest configurations as YAML objects."""
-    validate_annotations(annotations),
+def gen_loadtest_configs(
+        base_config: Mapping[str, Any],
+        scenario_name_regex: str,
+        language_config: scenario_config_exporter.LanguageConfig,
+        loadtest_name_prefix: str,
+        uniquifiers: Iterable[str],
+        annotations: Mapping[str, str],
+        runs_per_test: int = 1) -> Iterable[Dict[str, Any]]:
+    """Generates LoadTest configurations for a given language config.
+
+    The LoadTest configurations are generated as YAML objects.
+    """
+    validate_annotations(annotations)
     prefix = loadtest_name_prefix or default_prefix()
+    cl = language_config.client_language or language_config.language
+    sl = language_config.server_language or language_config.language
+    scenario_filter = scenario_config_exporter.scenario_filter(
+        scenario_name_regex=scenario_name_regex,
+        category=language_config.category,
+        client_language=language_config.client_language,
+        server_language=language_config.server_language)
+    scenarios = scenario_config_exporter.gen_scenarios(language_config.language,
+                                                       scenario_filter)
+
     for scenario in scenarios:
         for run_index in gen_run_indices(runs_per_test):
             uniq = uniquifiers + [run_index] if run_index else uniquifiers
             name = loadtest_name(prefix, scenario['name'], uniq)
             scenario_str = json.dumps({'scenarios': scenario}, indent='  ')
 
             config = copy.deepcopy(base_config)
+
             metadata = config['metadata']
             metadata['name'] = name
             if 'labels' not in metadata:
                 metadata['labels'] = dict()
+            metadata['labels']['language'] = language_config.language
             metadata['labels']['prefix'] = prefix
             if 'annotations' not in metadata:
                 metadata['annotations'] = dict()
             metadata['annotations'].update(annotations)
             metadata['annotations'].update({
                 'scenario': scenario['name'],
-                'uniquifiers': uniq,
+                'uniquifier': '-'.join(uniq),
             })
+
+            clients = config['spec']['clients']
+            clients.clear()
+            clients.extend((client for client in base_config['spec']['clients']
```

Comment:
the fact that you go through all the client specs in the template and select the one with the correct language isn't fully clear from the structure of the template or from the code. Perhaps we should add a comment?
"The base configuration obtained from the template has client specs for all possible languages. We select the one that matches the language of the scenario we are currently processing".

Also, it seems that if the  client spec for the given language is missing, the script will end up with an empty client spec for given scenario, without warning us about it?

Reply:
Added comment, and added a check for when the client or server is not found. Good catch, thanks!